doc/message-registry.md

```diff
@@ -0,0 +1,253 @@
+# Error index {#error-index}
+
+<!--
+NOTE: This file has been generated automatically by “update-message-registry.py”.
+      Do not edit manually!
+-->
+
+This page lists the warnings and errors generated by xkbcommon.
+There are currently 15 entries.
+
+@todo The documentation of the log messages is a work in progress.
+
+## Index
+
+| Code      | Identifier                   | Description | Type |
+| --------- | ---------------------------- | ----------- | ---- |
+| [XKB-107] | `unrecognized-keysym` | Warn on unrecognized keysyms. | Warning |
```

Comment:
What is the logic behind the numbering? I mean why is it not 001, 002 etc.?

Reply:
No and this is on purpose. I left a note in `doc/message-registry.yaml` with this link: [Code Assignment Recommendations](https://github.com/haskellfoundation/error-message-index/blob/main/tool-developers.md#code-assignment-recommendations).

I like the reasoning, but of course this is one point we have not discussed yet.

Reply:
Documented in registry & commit